kvcoord: move NodeDescScore to kvclient

andy-kimball opened the pull request:
The `kvcoord.NodeDescStore` interface is used in a number of other packages. Move it to the `kvclient` package in order to prevent a dependency cycle with the tenantcostclent package, which will soon need to use it.

Move `revision_reader.go` to the `backupccl` package. Besides a SQL test, the `backupccl` package is the only user of the `GetAllRevisions` function. This function uses the `storage` package, which is prohibited for use by the `kvcoord` package and now transitively by the `kvclient` package.

Epic: https://cockroachlabs.atlassian.net/browse/CC-28471

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127172)
<!-- Reviewable:end -->

andy-kimball committed:
kvcoord: move NodeDescScore to kvclient

The `kvcoord.NodeDescStore` interface is used in a number of other packages.
Move it to the `kvclient` package in order to prevent a dependency cycle with
the tenantcostclent package, which will soon need to use it.

Move `revision_reader.go` to the `backupccl` package. Besides a SQL test, the
`backupccl` package is the only user of the `GetAllRevisions` function. This
function uses the `storage` package, which is prohibited for use by the
`kvcoord` package and now transitively by the `kvclient` package.

Epic: https://cockroachlabs.atlassian.net/browse/CC-28471

Release note: None

andrewbaptist reviewed (approved):
LGTM!

Thanks for cleaning this up.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127172#-:-O1vrNWACMDSqqbzykXE:b-wyubda)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andy-kimball and @dt)

___
*[`pkg/ccl/backupccl/revision_reader_test.go` line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127172#-O1vqonK72IONpupXfuN:-O1vqonK72IONpupXfuO:b-q27dqn) ([raw file](https://github.com/cockroachdb/cockroach/blob/1c5ce343ad21642fa6e0935672d9e46acfbed1b9/pkg/ccl/backupccl/revision_reader_test.go#L1)):*
> ```Go
> // Copyright 2021 The Cockroach Authors.
> ```

nit: I'm not sure if the year matters, but consider leaving this as 2023.


<!-- Sent from Reviewable.io -->

andy-kimball commented:
bors r+
